src/test/javascript/spec/component/course/course-management-exercises-search.component.spec.ts

```diff
@@ -65,14 +44,14 @@ describe('Course Management Exercises Search Component', () => {
     });
 
     it('should change filter on name change', () => {
+        const emitSpy = jest.spyOn(comp.exerciseFilter, 'emit');
```

Comment:
Sorry, I wasn't clear enough :D
I meant moving it up the hierarchy so just in the before each.

If I think about it, we don't even need the restoreAllMocks for it since it gets reinitialised in the beforeEach then.

Reply:
=> Then remove it from the other tests accordingly